### PR TITLE
New tests for WIDERFace dataset

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR ports the WIDERFace tests to the new test infrastructure.

Addresses part of #3531

### CODE REVIEW

File: torchvision/datasets/widerface.py
Comment (NicolasHug) [inline torchvision/datasets/widerface.py:175]: That's not great, but for now `DatasetTestCase` cannot patch the `_check_integrity` method, so I had to move it as a function.

What would you recommend here @pmeier? I would tend towards updating `DatasetTestCase._patch_checks()` instead, as it's a bit unfortunate to have to change some code here just so that it's compatible with the test infrastructure

Comment (pmeier) [reply torchvision/datasets/widerface.py:175]: Why do you need to patch the `_verify_integrity` method? Shouldn't this simply return `True` if the fake data is correctly set up?

**Review by fmassa — APPROVED** [review]
LGTM thanks!

@pmeier I'll merge once you approve it

File: test/test_datasets.py
Comment (pmeier) [inline test/test_datasets.py:479]: No need to create this dictionary inside the loop.

File: test/test_datasets.py
Comment (pmeier) [inline test/test_datasets.py:493]: Same as above.

File: test/test_datasets.py
Comment (pmeier) [inline test/test_datasets.py:498]: Same as above with the caveat that we need `split_idx` here. Maybe also oust this and provide format strings?

```python
annotation_content[split].format(split_index)
```

File: test/test_datasets.py
Comment (pmeier) [inline test/test_datasets.py:481]: Instead of hard-coding this you could do something like `num_examples = split_idx`. That way we can be sure that the dataset is loading the correct data given by `split`. If all splits have the same length, we can't use that as indicator.

**Review by pmeier — COMMENTED** [review]
Some nitpicks. Otherwise LGTM. Thanks @NicolasHug!

File: test/test_datasets.py
Comment (NicolasHug) [reply test/test_datasets.py:496]: I moved the `split_to_idx` out of the loop as it's needed in different places, but for this one it seems more natural to me to leave it here. Moving it out would mean declaring a new variable that's only used once (here), and I tend to prefer to declare things where they're used

Comment (pmeier) [reply test/test_datasets.py:496]: No strong opinion here. I'm aware that this completely up the alley of premature optimization, but for me it _feels_ wrong to declare this dictionary in every step and then simply select a single value.

**Pull request merged**
